session/send.go

```diff
@@ -184,25 +237,21 @@ func (s *Session) composeMessage(recipient, provider string, message []byte, que
 		Payload:   payload,
 		WithSURB:  true,
 	}
-	if query {
-		msg.SURBType = cConstants.SurbTypeKaetzchen
-	} else {
-		msg.SURBType = cConstants.SurbTypeACK
-	}
+	msg.SURBType = cConstants.SurbTypeKaetzchen
 	return &msg, nil
 }
 
-// SendQuery sends a mixnet provider-side service query.
-func (s *Session) SendUnreliableQuery(recipient, provider string, message []byte) (*[cConstants.MessageIDLength]byte, error) {
-	msg, err := s.composeMessage(recipient, provider, message, true)
+// SendUnreliableMessage sends message without any automatic retransmissions.
+func (s *Session) SendUnreliableMessage(recipient, provider string, message []byte) (*[cConstants.MessageIDLength]byte, error) {
+	msg, err := s.composeMessage(recipient, provider, message)
 	if err != nil {
 		return nil, err
 	}
 
 	s.mapLock.Lock()
-	s.replyNotifyMap[*msg.ID] = new(sync.Mutex)
-	s.replyNotifyMap[*msg.ID].Lock()
 	s.messageIDMap[*msg.ID] = msg
+	s.waitChans[*msg.ID] = channels.NewInfiniteChannel()
```

Comment:
Why use an InfiniteChannel vs using native channels? Isn't there going to be only one reader/writer and one message per msg channel?